x-pack/legacy/plugins/index_management/public/components/template_form/template_steps.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiStepsHorizontal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  currentStep: number;
+  maxCompletedStep: number;
+  updateCurrentStep: (step: number, shouldValidate?: boolean) => void;
+  isStepValid: boolean;
+}
+
+const stepNamesMap: { [key: number]: string } = {
+  1: i18n.translate('xpack.idxMgmt.templateForm.steps.logisticsStepName', {
+    defaultMessage: 'Logistics',
+  }),
+  2: i18n.translate('xpack.idxMgmt.templateForm.steps.settingsStepName', {
+    defaultMessage: 'Index settings',
+  }),
+  3: i18n.translate('xpack.idxMgmt.templateForm.steps.mappingsStepName', {
+    defaultMessage: 'Mappings',
+  }),
+  4: i18n.translate('xpack.idxMgmt.templateForm.steps.aliasesStepName', {
+    defaultMessage: 'Aliases',
+  }),
+  5: i18n.translate('xpack.idxMgmt.templateForm.steps.summaryStepName', {
+    defaultMessage: 'Review template',
+  }),
+};
+
+export const TemplateSteps: React.FunctionComponent<Props> = ({
+  currentStep,
+  maxCompletedStep,
+  updateCurrentStep,
+  isStepValid,
+}) => {
+  const steps = [1, 2, 3, 4, 5].map(step => {
+    return {
+      title: stepNamesMap[step],
+      isComplete: maxCompletedStep >= step,
+      isSelected: currentStep === step,
+      disabled: isStepValid ? false : step > currentStep + 1,
```

Comment:
I think this shoul be

```js
disabled: isStepValid ? false : step !== currentStep, // disable all step but the current one
onClick: () => updateCurrentStep(step), // no more "shouldValidate" needed from my previous comment
```

Reply:
i was thinking about this some more - not sure that we really need to disable any of the steps, since we're always validating. WDYT? I removed it for now and also updated the `onClick`.

Reply:
Yeah, both can work. The fact that they are disabled brings the focus to the form. We can ask another pair of eyes on it cc @cjcenizal

Reply:
I'm not sure this works. If the first step is in an invalid state then I feel like all of the subsequent steps should be disabled, right?

![image](https://user-images.githubusercontent.com/1238659/63555436-bf7a3680-c4f5-11e9-88a5-dd91f441939d.png)

Reply:
@cjcenizal  Yes, this is what

```js
disabled: isStepValid ? false : step !== currentStep, // disable all step but the current one
```

does. It disables all the steps once 1 step is invalid.